File: client/src/types/Session.ts
import { User } from './User';
import { Song } from './Song';

export class Session {
    id: string
    name: string
    hostName: string
    inviteLink: string
    guests: {[key: string]: User}
    playlist: Song[]
    creationDate: Date
    isRunning: boolean

    constructor(data: {
        id: string
        name: string
        hostName: string
        inviteLink: string
        guests:  {[key: string]: User}
        playlistName: string
        playlist: Song[]
        creationDate: Date
        isRunning: boolean
    }) {
        this.id = data.id
        this.name = data.name
        this.hostName = data.hostName
        this.inviteLink = data.inviteLink
        this.guests = data.guests
        this.playlist = data.playlist
        this.creationDate = data.creationDate
        this.isRunning = data.isRunning ?? true
    }
}

File: server/app/models/session.py
from datetime import datetime
from typing import Optional

from .camel_model import CamelModel
from .user import User
from .recommendation import Recommendation
from .song import Playlist


class SessionCore(CamelModel):
    id: Optional[str] = None
    name: str
    host_id: Optional[str] = None
    host_name: Optional[str] = None
    guests: dict[str, User] = {}
    invite_link: Optional[str] = None
    creation_date: Optional[datetime] = None
    # is_running: bool = False


class Session(SessionCore):
    playlist: Playlist
    recommendations: list[Recommendation] = []